apps/dashboard/src/components/TrackingEvents.tsx

```diff
@@ -0,0 +1,47 @@
+// components/TrackingEvents.tsx
+import { TrackingEvent, TrackingStatus } from "@karrio/types/rest/api";
+import { formatDayDate } from "@karrio/lib";
+
+type DayEvents = { [k: string]: TrackingEvent[] };
+
+const computeEvents = (tracker: TrackingStatus): DayEvents => {
+  return (tracker?.events || []).reduce((days, event: TrackingEvent) => {
+    const daydate = formatDayDate(event.date as string);
+    return { ...days, [daydate]: [...(days[daydate] || []), event] };
+  }, {} as DayEvents);
```

Comment:
Doing it different... we can directly mutate the accumulator object instead of creating new objects at each iteration. The spread operator creates a new object and array during each iteration, which can lead to performance issues in larger datasets.